packages/react/src/components/GroupedList/GroupedListV2.base.tsx

```diff
@@ -0,0 +1,604 @@
+        return renderShowAll(item, flattenedIndex);
+      } else if (item.type === 'footer') {
+        return renderFooter(item, flattenedIndex);
+      } else if (item.type === 'item') {
+        const level = item.group.level ? item.group.level + 1 : 1;
+        return onRenderCell(level, item.item, item.itemIndex ?? flattenedIndex);
+      }
+    },
+    [onRenderCell, renderHeader, renderShowAll, renderFooter],
+  );
+
+  return (
+    <FocusZone
+      direction={FocusZoneDirection.vertical}
+      data-automationid="GroupedList"
+      data-is-scrollable="false"
+      role="presentation"
+      {...focusZoneProps}
+      shouldEnterInnerZone={shouldEnterInnerZone}
+      className={css(classNames.root, focusZoneProps.className)}
+    >
+      <List
+        ref={listRef}
+        role={role}
+        items={listView}
+        onRenderCellConditional={renderItem}
+        usePageCache={usePageCache}
+        onShouldVirtualize={onShouldVirtualize}
+        getPageSpecification={getPageSpecification}
+        version={version}
+        getKey={getKey}
+        {...rootListProps}
+      />
+    </FocusZone>
+  );
+};
+
+interface IGroupItemProps<T> {
+  props: T;
+  render: IRenderFunction<T>;
+  defaultRender: (props?: T) => JSX.Element | null;
+  item: any;
+  selection?: ISelection;
+  eventGroup?: EventGroup;
+}
+
+const GroupItem = <T,>({
+  render,
+  defaultRender,
+  item,
+  selection,
+  eventGroup,
+  props,
+}: React.PropsWithChildren<IGroupItemProps<T>>): React.ReactElement | null => {
+  const group = item.group;
+
+  const isSelected = useIsGroupSelected(group.startIndex, group.count, selection, eventGroup);
+  const mergedProps = {
+    ...props,
+    isSelected: isSelected,
+    selected: isSelected,
+  };
+  return render(mergedProps, defaultRender);
+};
+
+export class GroupedListV2Wrapper
+  extends React.Component<IGroupedListProps, IGroupedListV2State>
+  implements IGroupedList {
+  public static displayName: string = 'GroupedListV2';
+  private _list = React.createRef<List>();
+
+  public static getDerivedStateFromProps(
+    nextProps: IGroupedListProps,
+    previousState: IGroupedListV2State,
+  ): IGroupedListV2State {
+    const { groups } = nextProps;
+
+    if (groups !== previousState.groups) {
+      return {
+        ...previousState,
+        groups,
+      };
+    }
+
+    return previousState;
+  }
+
+  constructor(props: IGroupedListProps) {
+    super(props);
+    initializeComponentRef(this);
+
+    const { listProps: { version = {} } = {}, groups } = props;
+    this.state = {
+      version,
+      groupExpandedVersion: {},
+      groups,
+    };
+  }
+
+  public scrollToIndex(index: number, measureItem?: (itemIndex: number) => number, scrollToMode?: ScrollToMode): void {
+    if (this._list.current) {
+      this._list.current.scrollToIndex(index, measureItem, scrollToMode);
+    }
+  }
+
+  public getStartItemIndexInView(): number {
+    return this._list.current!.getStartItemIndexInView() || 0;
```

Comment:
You're checking `if (this._list.current)` in the function above -- do you need to do that here, instead of using the non-null assertion `!`